hybrasyl/Objects/User.cs

```diff
@@ -1155,12 +1167,20 @@ internal void UseSpell(byte slot, uint target = 0)
         /// <returns>True or false depending on success.</returns>
         public bool ProcessCastingCost(Castable castable)
         {
-            if (castable.CastCosts?.CastCost == null) return true;
+            if (castable.CastCosts.Count == 0) return true;
+
+            var costStruct = castable.CastCosts.Where(e => e.Class.Contains((Class)Class));
+
+            if (costStruct.Count() == 0)
+                costStruct = castable.CastCosts.Where(e => e.Class.Count == 0);
+
+            if (costStruct.Count() == 0)
+                return true;
 
-            var costs = castable.CastCosts.CastCost;
             uint reduceHp = 0;
             uint reduceMp = 0;
-            int removeNumItems = 0;
+            bool hasItemCost = true;
+            var costs = costStruct.First();

```

Comment:
this will only apply the first cast cost, and is handled improperly.

Reply:
The cast cost enumerated is either the first one that specifies a class, or a default (no class). Why would you need to process multiple cast cost structures for the same class? There is no need for that duplication.

Reply:
Fair enough. This should be moved to the above comment. as .First() at the end for clarity.